lab_5_scrapper/scrapper.py

```diff
@@ -26,11 +91,50 @@ def _extract_config_content(self) -> ConfigDTO:
         Returns:
             ConfigDTO: Config values
         """
+        with open(self.path_to_config, 'r', encoding='utf-8') as file:
+            config = json.load(file)
+        return ConfigDTO(
+            seed_urls=config["seed_urls"],
+            total_articles_to_find_and_parse=config["total_articles_to_find_and_parse"],
+            headers=config["headers"],
+            encoding=config["encoding"],
+            timeout=config["timeout"],
+            should_verify_certificate=config["should_verify_certificate"],
+            headless_mode=config["headless_mode"]
+        )
 
     def _validate_config_content(self) -> None:
         """
         Ensure configuration parameters are not corrupt.
         """
+        with open(self.path_to_config, 'r', encoding='utf-8') as file:
+            conf = json.load(file)
+
+        if not (isinstance(conf['seed_urls'], list)
+                and all(re.match(r"https?://(www.)?", seed_url)
+                        for seed_url in conf['seed_urls']
+                        )
+                ):
+            raise IncorrectSeedURLError
+
+        num = conf['total_articles_to_find_and_parse']
+        if not isinstance(num, int) or num < 0 or isinstance(num, bool):
+            raise IncorrectNumberOfArticlesError
+        if isinstance(num, int) and not 1 <= num <= 150:
+            raise NumberOfArticlesOutOfRangeError
+
+        if not isinstance(conf['headers'], dict):
+            raise IncorrectHeadersError
+
+        if not isinstance(conf['encoding'], str):
+            raise IncorrectEncodingError
+
+        if not isinstance(conf['should_verify_certificate'], bool) \
+                or not isinstance(conf['headless_mode'], bool):
```

Comment:
headless should not be checked. if it is not stated in docs (is it?)

Reply:
It's not in the documentation

But ther's a test on this _(s2_1_crawler_config_test._test_incorrect_headless_config_param)_

Reply:
ok, I see